addons/website_sale_stock/models/sale_order.py

```diff
@@ -41,14 +41,25 @@ def _verify_updated_quantity(self, order_line, product_id, new_qty, **kwargs):
                         order_line._set_shop_warning_stock(total_cart_qty, available_qty)
                     else:
                         self._set_shop_warning_stock(total_cart_qty, available_qty)
+                    returned_warning = order_line.shop_warning or self.shop_warning
                 else:  # 0 or negative allowed_qty
                     # if existing line: it will be deleted
                     # if no existing line: no line will be created
-                    self.shop_warning = _(
-                        "Some products became unavailable and your cart has been updated. We're sorry for the inconvenience.")
-                return allowed_line_qty, order_line.shop_warning or self.shop_warning
+                    returned_warning = _(
+                        "The item has not been added to your cart since it is not available."
+                    )
+                return allowed_line_qty, returned_warning
         return super()._verify_updated_quantity(order_line, product_id, new_qty, **kwargs)
 
+    def _cart_update_order_line(self, product_id, quantity, order_line, **kwargs):
+        res = super()._cart_update_order_line(product_id, quantity, order_line, **kwargs)
+        if order_line and quantity <= 0:
+            self.shop_warning = _(
+                "Some products became unavailable and your cart has been updated. We're sorry for "
+                "the inconvenience."
+            )
+        return res
```

Comment:
I don't really like this change, it's not restricted to the stock logic and might replace the warning in other situations depending on other potential overrides of `_verify_update_quantity`